app/plugins/Files/js/loaders.js

```diff
@@ -59,6 +59,37 @@ addResultListener('uploaded', function(result) {
 	update();
 });
 
+// Non-recursively upload all files in a directory
+function uploadDir(dirPath, nickname) {
+	var fs = require("fs");
+	var path = require("path");
+
+	// Upload files one at a time
+	fs.readdir(dirPath, function(err, files) {
+		if (err) {
+			notify('Failed retrieving directory contents', 'error');
+			return;
+		}
+		files.forEach(function(file) {
+			try {
+				var filePath = path.join(dirPath, file);
+
+				// Check that file exists
+				var stats = fs.statSync(filePath);
+
+				// Skip hidden files and directories
+				if (~isUnixHiddenPath(filePath) & stats.isFile()) {
+					upload(filePath, nickname + file);
+				}
+			}
+			catch (e) {
```

Comment:
You should use [`fs.stat`](https://nodejs.org/api/fs.html#fs_fs_stat_path_callback) which holds two arguments, a file path and a callback which in turn holds an error and your stats variable, e.g.
```
fs.stat(filePath, function(err, stats) {
    if (err) {
        // handle it
        return;
    }
    // do stuff with stats
}
```